### PR TITLE
topline might be changed in diff mode unexpectedly

### LINKED ISSUE #15812: line("w0") breaks diff mode alignment
### Steps to reproduce

Run vim as follows
```
vim --clean -S <(tee << 'EOF'
    set lines=20
    call setline(1, range(1,60))
    vnew
    call setline(1, range(1,10) + range(50,60))
    windo diffthis
    norm! G
    exe "norm! 30\<C-y>"
EOF
)
```
image

Then, run `:echo line('w0', 1001)` (1001 is the left window)
image
Note that filler lines are no longer properly aligned.

### Expected behaviour

The alignment stays intact. 

### Version of Vim

9.1.759

### Environment

ubuntu 22.04

### Logs and stack traces

_No response_

### PR DESCRIPTION
**Pull request by chrisbra**
Problem:  topline might be changed in diff mode unexpectedly
Solution: do not re-calculate topline, when using line() func
          in diff mode.

fixes: #15812

### CODE REVIEW

File: src/evalfunc.c [hunk 1]
```diff
@@ -8359,9 +8359,15 @@ f_line(typval_T *argvars, typval_T *rettv)
 	{
 	    if (switch_win_noblock(&switchwin, wp, tp, TRUE) == OK)
 	    {
+		// in diff mode, prevent that the window scrolls
+		// and keep the topline
+		if (curwin->w_p_diff && switchwin.sw_curwin->w_p_diff)
```
Comment (zeertzjq) [inline src/evalfunc.c:8364]: Why is `w_p_diff` from `sw_curwin` (i.e. the old `curwin`) also checked?

Comment (chrisbra) [reply src/evalfunc.c:8364]: Because we only want to disable updating topline, if both buffers are in diff mode

Comment (zeertzjq) [reply src/evalfunc.c:8364]: What two are "both" buffers? If there are 3 windows (say `1000`, `1001`, `1002`) with 2 (say `1001` and `1002`) in diff mode, should the behavior of calling `line('w0', 1002)` be different depending on whether the current window is `1000` or `1001`?

Comment (chrisbra) [reply src/evalfunc.c:8364]: I went with the minimal approach, that changes the behaviour only if we are temporarily switching from one diff buffer to another diff buffer. And only then I think updating the topline could be skipped.

But if we are either switching from a non-diff buffer to a diff buffer (or from a diff buffer to a non-diff buffer), I think the topline should still be updated.

**Pull request closed without merging**